src/common/consts.rs

```diff
@@ -20,81 +49,85 @@ pub enum HitboxVisualization {
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum Direction {
     None = 0,
-    Right = 1,
-    UpRight = 2,
-    Up = 3,
-    UpLeft = 4,
-    Left = 5,
-    DownLeft = 6,
-    Down = 7,
-    DownRight = 8,
+    Right = 0x1,
+    UpRight = 0x2,
+    Up = 0x4,
+    UpLeft = 0x8,
+    Left = 0x10,
+    DownLeft = 0x20,
+    Down = 0x40,
+    DownRight = 0x80,
     // lol what goes here jug smh my head
-    Random = 9,
+    Random = 0x100,
```

Comment:
Just a suggestion, maybe use bit shifting so this code is easier to read? 
```rust
pub enum Direction {
    None = 0,
    Right = 1 << 0,
    UpRight = 1 << 1,
    Up = 1 << 2,
    UpLeft = 1 << 3,
    Left = 1 << 4,
    DownLeft = 1 << 5,
    Down = 1 << 6,
    DownRight = 1 << 7,
    // lol what goes here jug smh my head
    Random = 1 << 8,
}
```

Reply:
Does it work with pattern matching?

Reply:
Ah, it won't, good point.